+Patch Description: Modified the tests to get group identity with root permissions.
diff --git a/testcases/kernel/syscalls/setfsgid/setfsgid03.c b/testcases/kernel/syscalls/setfsgid/setfsgid03.c
index dd182318b..dd3b4b256 100644
--- a/testcases/kernel/syscalls/setfsgid/setfsgid03.c
+++ b/testcases/kernel/syscalls/setfsgid/setfsgid03.c
@@ -35,6 +35,7 @@

 TCID_DEFINE(setfsgid03);
 int TST_TOTAL = 1;
+gid_t gid;

 static char nobody_uid[] = "nobody";
 static struct passwd *ltpuser;
@@ -46,8 +47,6 @@ int main(int ac, char **av)
 {
        int lc;

-       gid_t gid;
-
        tst_parse_opts(ac, av, NULL, NULL);

        setup();
@@ -55,10 +54,6 @@ int main(int ac, char **av)
        for (lc = 0; TEST_LOOPING(lc); lc++) {
                tst_count = 0;

-               gid = 1;
-               while (!getgrgid(gid))
-                       gid++;
-
                GID16_CHECK(gid, setfsgid, cleanup);

                TEST(SETFSGID(cleanup, gid));
@@ -86,6 +81,10 @@ int main(int ac, char **av)
 static void setup(void)
 {
        tst_require_root();
+
+       gid = 1;
+       while (!getgrgid(gid))
+               gid++;

        ltpuser = getpwnam(nobody_uid);
        if (ltpuser == NULL)

